osu.Game.Modes.Taiko/Objects/Drawable/DrawableTaikoHitObject.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Game.Modes.Objects.Drawables;
+using osu.Game.Modes.Taiko.Judgements;
+
+namespace osu.Game.Modes.Taiko.Objects.Drawable
+{
+    public class DrawableTaikoHitObject : DrawableHitObject<TaikoHitObject, TaikoJudgementInfo>
+    {
+        /// <summary>
+        /// The colour used for various elements of this DrawableHitObject.
+        /// </summary>
+        public virtual Color4 AccentColour { get; }
+
+        public DrawableTaikoHitObject(TaikoHitObject hitObject)
+            : base(hitObject)
+        {
+            Anchor = Anchor.CentreLeft;
+            Origin = Anchor.Centre;
+
+            RelativePositionAxes = Axes.X;
+
+            LifetimeStart = HitObject.StartTime - HitObject.PreEmpt * 2;
+            LifetimeEnd = HitObject.StartTime + HitObject.PreEmpt;
+
+            // Todo: Remove (suppresses Resharper)
+            AccentColour = Color4.White;
+        }
+
+        protected override TaikoJudgementInfo CreateJudgementInfo() => new TaikoJudgementInfo();
+
+        protected override void UpdateState(ArmedState state)

```

Comment:
maybe we want to make the base `UpdateState` empty-virtual to avoid this kinda thing.